block/internal/syncing/syncer.go

```diff
@@ -682,42 +697,105 @@ func (s *Syncer) verifyForcedInclusionTxs(currentState types.State, data *types.
 		return fmt.Errorf("failed to retrieve forced included txs from DA: %w", err)
 	}
 
-	// If no forced inclusion transactions found, nothing to verify
-	if len(forcedIncludedTxsEvent.Txs) == 0 {
-		s.logger.Debug().Uint64("da_height", currentState.DAHeight).Msg("no forced inclusion transactions to verify")
-		return nil
-	}
-
+	// Build map of transactions in current block
 	blockTxMap := make(map[string]struct{})
 	for _, tx := range data.Txs {
 		blockTxMap[hashTx(tx)] = struct{}{}
 	}
 
-	// Check if all forced inclusion transactions are present in the block
-	var missingTxs [][]byte
+	// Check if any pending forced inclusion txs from previous epochs are included
+	var stillPending []pendingForcedInclusionTx
+	s.pendingForcedInclusionTxs.Range(func(key, value interface{}) bool {
+		pending := value.(pendingForcedInclusionTx)
+		if _, ok := blockTxMap[pending.TxHash]; ok {
+			s.logger.Debug().
+				Uint64("height", data.Height()).
+				Uint64("epoch_start", pending.EpochStart).
+				Uint64("epoch_end", pending.EpochEnd).
+				Str("tx_hash", pending.TxHash[:16]).
+				Msg("pending forced inclusion transaction included in block")
+			s.pendingForcedInclusionTxs.Delete(key)
+		} else {
+			stillPending = append(stillPending, pending)
+		}
+		return true
+	})
+
+	// Add new forced inclusion transactions from current epoch
```

Comment:
we should make sure to document this change in the adr. Nice job on the change!!